server.py

```diff
@@ -118,24 +177,48 @@ def get_verified_email(token):
     except Exception:
         return {'error': 'Invalid token'}
 
+    # Take the email address from the 'subject' claim in the payload, and
+    # verify the format of the email address is what we expect.
     sub = payload['sub']
     if not re.match('.+@.+', sub):  # <-- TODO: Use a proper parser.
```

Comment:
I think we should probably get rid of the TODO here. At most, we can check for the rules used in the `emailaddress` Rust crate we're using for the broker, where I think it also ends up using a totally simple algorithm for this in the end, because it's most web-compatible and trying to parse email addresses exactly right is just not worth it in terms of complexity.